R/Model-methods.R

```diff
@@ -1,124 +1,124 @@
-##' @describeIn dose Compute the doses for a given probability, given
-##' Pseudo DLE model with samples
-##' @example examples/Model-method-dose-modelTox.R
-setMethod("dose",
-          signature=
-            signature(prob="numeric",
-                      model="ModelTox",
-                      samples="Samples"),
-          def=
-            function(prob, model, samples, ...){
-              ## extract the dose function from the model
-              doseFun <- slot(model, "dose")
-              ## which arguments, besides the prob, does it need?
-              argNames <- setdiff(names(formals(doseFun)),
-                                  "prob")
-              ## now call the function with prob and with
-              ## the arguments taken from the samples
-              ret <- do.call(doseFun,
-                             c(list(prob=prob),
-                               samples@data[argNames]))
-              ## return the resulting vector
-              return(ret)
-            })
-
-## =========================================================================
-## ----------------------------------------------------------------------------
-## Compute the dose for a given Pseudo DLE model and a given probability
-## -----------------------------------------------------------------------
-##' @describeIn dose Compute the dose for a given probability and a given
-##' Pseudo DLE model without samples
-##' @example examples/Model-method-doseNoSamples.R
-setMethod("dose",
-          signature=
-            signature(prob="numeric",
-                      model="ModelTox",
-                      samples="missing"),
-          def=
-            function(prob, model, ...){
-              ## extract the dose function from the model
-              doseFun <- slot(model, "dose")
-              ## which arguments, besides the prob, does it need?
-              argNames <- setdiff(names(formals(doseFun)),
-                                  "prob")
-              values<-c()
-              for (parName in argNames){
-                values<-c(values, slot(model,parName))}
-              ## now call the function with prob
-              ret <- do.call(doseFun,
-                             c(list(prob=prob), values))
-              ## return the resulting vector
-              return(ret)
-            })
-
+#' @include Model-class.R Samples-class.R
+NULL
+
+# dose ----
+
+#' Computing the Doses for a Given Probability, Model and the Samples
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' @param prob (`number`)\cr the toxicity probability which is targeted.
+#' @param model (`Model` or `ModelTox`)\cr the model for single agent dose
+#'   escalation or pseudo DLE (dose-limiting events)/toxicity model.
+#' @param samples (`Samples`)\cr the samples that will be used to compute the
+#'   resulting dose by `model@dose` function.
+#' @param ... not used.
+#' @return A `numeric` output from the `model@dose` function.
```

Comment:
I would explain a bit here that the returned values have one number for each sample.